torch_xla/csrc/tensor.cpp

```diff
@@ -242,15 +242,15 @@ void XLATensor::SetShardingSpec(const ShardingSpec& sharding) {
         << ", must be cleared before applying a new one, "
         << sharding.sharding.DebugString();
   }
-  dynamic_cast<XlaNode*>(GetIrValue().node.get())
-      ->SetSharding(sharding_spec()->sharding);
+  CreateShardedIrValue(sharding_spec());
```

Comment:
Nit, `CreateXXX` sounds like we will return something, but this method actually inplace update the `data()`. Let's maybe call it `SetShardedIrValue` instead?

Reply:
Done